components/event-bus/internal/knative/subscription/controller/eventactivation/reconcile.go

```diff
@@ -3,114 +3,131 @@ package eventactivation
-		Requeue: requeue,
-	}, reconcileErr
+
+	return reconcileErr
 }
 
-func (r *reconciler) reconcile(ctx context.Context, ea *eventingv1alpha1.EventActivation) (bool, error) {
+// eventActivationByKey retrieves a EventActivation object from a lister by ns/name key.
+func eventActivationByKey(key string, l applicationconnectorlistersv1alpha1.EventActivationLister) (*applicationconnectorv1alpha1.EventActivation, error) {
+	ns, name, err := cache.SplitMetaNamespaceKey(key)
+	if err != nil {
+		return nil, controller.NewPermanentError(pkgerrors.Wrap(err, "invalid object key"))
+	}
+
+	src, err := l.EventActivations(ns).Get(name)
+	switch {
+	case apierrors.IsNotFound(err):
+		return nil, controller.NewPermanentError(pkgerrors.Wrap(err, "object no longer exists"))
+	case err != nil:
+		return nil, err
+	}
 
+	return src, nil
+}
+
+func (r *Reconciler) reconcile(ctx context.Context, ea *applicationconnectorv1alpha1.EventActivation) error {
+	log := logging.FromContext(ctx)
 	// delete or add finalizers
 	if !ea.DeletionTimestamp.IsZero() {
 		// deactivate all Kyma subscriptions related to this ea
-		subs, _ := util.GetSubscriptionsForEventActivation(ctx, r.client, ea)
-		util.DeactivateSubscriptions(ctx, r.client, subs, log, r.time)
-
+		subs, err := util.GetSubscriptionsForEventActivation(r.kymaEventingClient, ea)
+		if err != nil {
+			return err
+		}
+		error := util.DeactivateSubscriptions(r.kymaEventingClient, subs, log, r.time)
+		if error != nil {
+			return error
+		}
 		// remove the finalizer from the list
 		ea.ObjectMeta.Finalizers = util.RemoveString(&ea.ObjectMeta.Finalizers, finalizerName)
-		log.Info("Finalizer removed", "Finalizer name", finalizerName)
-		return false, nil
+		log.Info("Finalizer removed", zap.String("Finalizer name", finalizerName))
+		return nil
 	}
 
 	// If we are adding the finalizer for the first time, then ensure that finalizer is persisted
 	if !util.ContainsString(&ea.ObjectMeta.Finalizers, finalizerName) {
 		//Finalizer is not added, let's add it
 		ea.ObjectMeta.Finalizers = append(ea.ObjectMeta.Finalizers, finalizerName)
-		log.Info("Finalizer added", "Finalizer name", finalizerName)
-		return true, nil
+		log.Info("Finalizer added", zap.String("Finalizer name", finalizerName))
+		return nil
 	}
 
-	// check an activate, if necessary, all the subscriptions
-	if subs, err := util.GetSubscriptionsForEventActivation(ctx, r.client, ea); err != nil {
-		log.Error(err, "GetSubscriptionsForEventActivation() failed")
-	} else {
-		log.Info("Kyma subscriptions found: ", "subs", subs)
-		// activate all subscriptions
-		if err := util.ActivateSubscriptions(ctx, r.client, subs, log, r.time); err != nil {
-			log.Error(err, "ActivateSubscriptions() failed")
-			return false, err
-		}
+	// check and activate, if necessary, all the subscriptions
+	subs, err := util.GetSubscriptionsForEventActivation(r.kymaEventingClient, ea)
+	if err != nil {
+		log.Error("GetSubscriptionsForEventActivation() failed", zap.Error(err))
```

Comment:
We could avoid those explicit calls to `log.Error` by simply wrapping returned errors with their context:
```diff
- log.Error("GetSubscriptionsForEventActivation() failed", zap.Error(err))
- return err
+ return pkgerrors.Wrap(err, "GetSubscriptionsForEventActivation() failed")
```
(the import is `pkgerrors "github.com/pkg/errors"`)